src/components/paginator.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+
+import { Dispatch } from '../lib/redirects-manager-state';
+
+type getPageNumbersType = (viewLength: number, perPage: number) => number[];
+
+type validateSelectedPageType = (input: string, max: number) => number;
+
+interface PaginatorComponentProps {
+    viewLength: number;
+    filterBy: string;
+    storeLength: number;
+    perPage: number;
+    currentPage: number;
+    dispatch: Dispatch;
+}
+
+interface PageSelectorProps { pageNumbers: number[]; current: number; dispatch: Dispatch; }
+
+const getPageNumbers: getPageNumbersType = (viewLength, perPage) => {
```

Comment:
please use inline types for functions, unless you really need to share interfaces ie. for callbacks

Reply:
applied